publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java

```diff
@@ -142,7 +142,27 @@ private static Resource convertToResource(Publication publication) {
                    .withDuplicateOf(publication.getDuplicateOf())
                    .withCuratingInstitutions(publication.getCuratingInstitutions())
                    .withImportDetails(publication.getImportDetails())
-                   .build();
+                   .build().validate();
+    }
+
+    private Resource validate() {
+        var cristinIdentifiers = getCristinIdentifiers();
+        if (hasMultipleEntries(cristinIdentifiers)) {
+            throw new MultipleCristinIdentifiersException();
+        } else {
+            return this;
+        }
```

Comment:
I'm not sure that this is really the case — are there zero duplicates in Cristin?

Reply:
When importing publications from Brage, multiple publications are ending up with IllegalArgumentException when we are trying to persist publication with multiple Cristin id's. There should not be duplicates in Cristin or Brage, but I think we have to handle cases when trying to persist publication with multiple Cristin identifiers.